experimental/QuadFormAndIsom/src/lattices_with_isometry.jl

```diff
@@ -1471,43 +1692,103 @@ julia> order(G)
+    fL = OL(f; check = false)
+    UL, _ = centralizer(OL, fL)
+    return image(discriminant_representation(L, UL; check = false))
+  elseif is_of_hermitian_type(Lf)
+    # Indefinite of rank bigger or equal to 3
     # We use Hermitian Miranda-Morrison (see [BH23, Part 6])
     dets, j = Oscar._local_determinants_morphism(Lf)
     _, jj = kernel(dets)
     jj = compose(jj, j)
     return image(jj)
+  else
+    # For this last case, we cut our lattice into two orthogonal parts and we
+    # proceed by induction by gluing the stabilizers. For now we do naively,
+    # and we split the "largest non trivial exponent of the isometry".
+    #
+    # TODO: make a "smart search", for instance for some particular stable
+    # kernel sublattices of small rank or for which rank(L) == euler_phi(n)
+    #
+    # We use the similar method as used for extending stabilizers along
+    # equivariant primitive extensions as seen in the function
+    # `admissible_equivariant_primitive_extensions`.
+    divs = sort!(divisors(n); rev = true)
+    k = divs[findfirst(k -> rank(kernel_lattice(Lf, k)) != 0, divs)]
+
+    M = kernel_lattice(Lf, k)
+    qM, fqM = discriminant_group(M)
+    GM, _ = image_centralizer_in_Oq(M)
+
+    N = orthogonal_submodule(Lf, basis_matrix(M))
+    qN, fqN = discriminant_group(N)
+    GN, _ = image_centralizer_in_Oq(N)
+
+    phi, HMinqM, HNinqN = glue_map(L, lattice(M), lattice(N); check = false)
+
+    # Since M and N are obtained by cutting some parts of `f \in O(L)`, the glue
+    # map should be equivariant!
+    @hassert :ZZLatWithIsom 1 is_invariant(fqM, HMinqM)
+    @hassert :ZZLatWithIsom 1 is_invariant(fqN, HNinqN)
+    @hassert :ZZLatWithIsom 1 matrix(compose(restrict_automorphism(fqM, HMinqM; check = false), phi)) == matrix(compose(phi, restrict_automorphism(fqN, HNinqN; check = false)))
+
+    HM = domain(HMinqM)
+    OHM = orthogonal_group(HM)
+
+    HN = domain(HNinqN)
+    OHN = orthogonal_group(HN)
+
+    _, qMinD, qNinD, _, OqMinOD, OqNinOD = _sum_with_embeddings_orthogonal_groups(qM, qN)
+    HMinD = compose(HMinqM, qMinD)
+    HNinD = compose(HNinqN, qNinD)
+
+    stabM, _ = stabilizer(GM, HMinqM)
+    stabN, _ = stabilizer(GN, HNinqN)
+
+    actM = hom(stabM, OHM, elem_type(OHM)[OHM(restrict_automorphism(x, HMinqM; check = false)) for x in gens(stabM)])
+    actN = hom(stabN, OHN, elem_type(OHN)[OHN(restrict_automorphism(x, HNinqN; check = false)) for x in gens(stabN)])
+
+    _, _, graph = _overlattice(phi, HMinD, HNinD, isometry(M), isometry(N); same_ambient = true)
+    disc, stab = _glue_stabilizers(phi, actM, actN, OqMinOD, OqNinOD, graph)
+    qL, fqL = discriminant_group(Lf)
+    OqL = orthogonal_group(qL)
+    phi = hom(qL, disc, TorQuadModuleElem[disc(lift(x)) for x in gens(qL)])
+    @hassert :ZZLatWithIsom 1 is_isometry(phi)
+    @hassert :ZZLatWithIsom 1 qL == disc
+
+    stab = sub(OqL, elem_type(OqL)[OqL(compose(phi, compose(g, inv(phi))); check = false) for g in stab])
```

Comment:
Is the output `stab` of `_glue_stabilizers` a group or a list of generators?
If it is a group, you can iterate over the generators only here.

Reply:
Those are single elements, but I might have to change the `_glue_stabilizers` function to return unique occurrences (I did not check that I could have several times a same element)